collection_requirements.rst

```diff
@@ -48,6 +48,51 @@ Collection Infrastructure
+
+- `controller-environment`: The plugins/modules always run in the same environment (Python interpreter, venv, host, etc) as ansible-core itself.
+- `other-environment`: It is possible, even if uncommon in practice, for the plugins/modules to run in a different environment than ansible-core itself.
+
+One ideal scenario where the `even if` clause comes into play is when using Cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which can't directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.
+
+**Controller-environment**
+
+In the controller environment, collections MUST support Python 2 (version 2.7) and Python 3 (Version 3.6 and higher), unless required libraries do not support these versions. Collections SHOULD also support Python v3.5 if all required libraries support this version. 
+
+**Other-environment**
+
+In the other environment, collections MUST support Python 2 (version 2.7) and Python 3 (Version 3.6 and higher), unless required libraries do not support these versions. Collections SHOULD also support Python v2.6 and v3.5 if all required libraries support this version.
+
+**Note**
+
+If the Collections does not support Python 2.6 and/or Python 3.5 explicitly then kindly take the below points into consideration:
+
+- Not supporting Python 2.6 on the managed node means that you are dropping support for RHEL6, which has extended support until 2024. 
+
+- Not supporting Python 3.5 means that Python 2.7 has to be installed on Ubuntu Xenial (16.04) and that you have to support Python 2.7.
+
+- Also, note that dropping support for a Python version for an existing module/plugin is a breaking change, and thus requires a major release. Hence, a collection MUST announce dropping support for Python versions in their changelog, if possible to announce in advance (i.e. in previous versions when support will be dropped).
+
+Standards and conventions while developing module utilities for collection
+--------------------------------------------------------------------------
+
+- `module_utils` can be marked for only internal use in the collection, but they MUST document this and MUST use a leading underscore for filenames.
+
+- It is a breaking change when you make an existing `module_utils` private and in that case the collection requires a major version bump.
+
+- Below are some recommendations for `module_utils` documentation: 
+1. no docstring = everything we recommend for `other-environment` is supported
+2. docstring 'Python versions supported: as for controller-environment' = everything we recommend for `controller-environment` is supported
+3. docstring with specific versions otherwise: 'Python versions supported: '
```

Comment:
```suggestion

  1. no docstring = everything we recommend for `other-environment` is supported
  2. docstring `'Python versions supported: as for controller-environment'` = everything we recommend for `controller-environment` is supported
  3. docstring with specific versions otherwise: `'Python versions supported: '`
```